src/Classes/PartyTab.lua

```diff
@@ -0,0 +1,620 @@
+				wipeTable(self.processedInput["Aura"])
+				self.processedInput["Aura"] = {}
+			end
+			if self.controls.importCodeMode.selIndex == 1 or self.controls.importCodeMode.selIndex == 3 then
+				self.controls.editCurses:SetText("")
+				wipeTable(self.processedInput["Curse"])
+				self.processedInput["Curse"] = {}
+			end
+			if self.controls.importCodeMode.selIndex == 1 or self.controls.importCodeMode.selIndex == 4 then
+				self.controls.enemyCond:SetText("")
+			end
+			if self.controls.importCodeMode.selIndex == 1 or self.controls.importCodeMode.selIndex == 5 then
+				self.controls.enemyMods:SetText("")
+			end
+			wipeTable(self.enemyModList)
+			self.enemyModList = new("ModList")
+			self.build.buildFlag = true 
+			return
+		end
+		local importCodeFetching = false
+		if self.importCodeSite and not self.importCodeXML then
+			self.importCodeFetching = true
+			local selectedWebsite = buildSites.websiteList[self.importCodeSite]
+			buildSites.DownloadBuild(self.controls.importCodeIn.buf, selectedWebsite, function(isSuccess, data)
+				self.importCodeFetching = false
+				if not isSuccess then
+					self.importCodeDetail = colorCodes.NEGATIVE..data
+					self.importCodeValid = false
+				else
+					importCodeHandle(data)
+					finishImport()
+				end
+			end)
+			return
+		end
+
+		finishImport()
+	end)
+	self.controls.importCodeGo.label = function()
+		return self.controls.importCodeMode2.selIndex == 3 and "Clear" or "Import"
+	end
+	self.controls.importCodeGo.enabled = function()
+		return (self.importCodeValid and not self.importCodeFetching) or self.controls.importCodeMode2.selIndex == 3
+	end
+	self.controls.importCodeGo.x = function()
+		return (self.width > 1350) and 8 or (-328)
+	end
+	self.controls.importCodeGo.y = function()
+		return (self.width > 1350) and 0 or 28
+	end
+	self.controls.importCodeGo.enterFunc = function()
+		if self.importCodeValid then
+			self.controls.importCodeGo.onClick()
+		end
+	end
+	
+	self.controls.rebuild = new("ButtonControl", {"LEFT",self.controls.importCodeGo,"RIGHT"}, 8, 0, 160, 20, "Rebuild All", function() 
+		wipeTable(self.processedInput)
+		wipeTable(self.enemyModList)
+		self.processedInput = { Aura = {}, Curse = {} }
+		self.enemyModList = new("ModList")
+		if self.controls.importCodeMode2.selIndex ~= 3 then
+			self:ParseBuffs(self.processedInput["Aura"], self.controls.editAuras.buf, "Aura")
+			self:ParseBuffs(self.processedInput["Curse"], self.controls.editCurses.buf, "Curse")
+			self:ParseBuffs(self.enemyModList, self.controls.enemyCond.buf, "EnemyConditions")
+			self:ParseBuffs(self.enemyModList, self.controls.enemyMods.buf, "EnemyMods")
+		end
+		self.build.buildFlag = true 
+	end)
+	self.controls.rebuild.label = function()
+		return self.controls.importCodeMode2.selIndex == 3 and "Remove effects" or "Rebuild All"
+	end
+	self.controls.rebuild.tooltipText = "^7Reparse all the inputs incase they have changed since loading the build or importing"
```

Comment:
Hide this when the button (if it's still needed per the above comment) changes function.

Reply:
again, if I change clear to a dedicated button this functionality is removed anyway, but can easily update this in the meantime